### PR TITLE
Automated PR: Cookstyle Changes

### PR DESCRIPTION
**Pull request by kitchen-porter**
Hey!
I ran Cookstyle 7.24.1 against this repo and here are the results.
This repo was selected due to the topics of chef-cookbook

### Issues found and resolved with libraries/helpers.rb

 - 127:10 refactor: `Chef/Modernize/UseChefLanguageSystemdHelper` - Chef Infra Client 15.5 and later include a `systemd?` helper for checking if a Linux system uses systemd. (https://docs.chef.io/workstation/cookstyle/chef_modernize_usecheflanguagesystemdhelper)
 - 272:7 refactor: `Chef/Modernize/UseChefLanguageSystemdHelper` - Chef Infra Client 15.5 and later include a `systemd?` helper for checking if a Linux system uses systemd. (https://docs.chef.io/workstation/cookstyle/chef_modernize_usecheflanguagesystemdhelper)

### Issues found and resolved with metadata.rb

 - 20:1 refactor: `Chef/Modernize/DependsOnChefVaultCookbook` - Don't depend on the chef-vault cookbook made obsolete by Chef Infra Client 16.0. The chef-vault gem and helpers are now included in Chef Infra Client itself. (https://docs.chef.io/workstation/cookstyle/chef_modernize_dependsonchefvaultcookbook)

### Issues found and resolved with recipes/service.rb

 - 95:32 refactor: `Chef/Modernize/UseChefLanguageSystemdHelper` - Chef Infra Client 15.5 and later include a `systemd?` helper for checking if a Linux system uses systemd. (https://docs.chef.io/workstation/cookstyle/chef_modernize_usecheflanguagesystemdhelper)

### Issues found and resolved with resources/splunk_app.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### Issues found and resolved with resources/splunk_index.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### Issues found and resolved with resources/splunk_installer.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### Issues found and resolved with resources/splunk_monitor.rb

 - 1:1 refactor: `Chef/Deprecations/ResourceWithoutUnifiedTrue` - Set `unified_mode true` in Chef Infra Client 15.3+ custom resources to ensure they work correctly in Chef Infra Client 18 (April 2022) when Unified Mode becomes the default. (https://docs.chef.io/workstation/cookstyle/chef_deprecations_resourcewithoutunifiedtrue)

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -269,7 +269,7 @@ def upgrade_enabled?
     end
 
     def systemd?
-      node['init_package'] == 'systemd'
+      systemd?
```
Comment (tas50) [inline libraries/helpers.rb:272]: This whole thing just needs to go away

File: resources/splunk_app.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@
 require 'pathname'
 
 provides :splunk_app
```
Comment (damacus) [inline resources/splunk_app.rb:19]: These can actually go if we rename the file to app.rb.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def create_data_bag_item(server, bag, item)
 RSpec.configure do |config|
   config.color = true               # Use color in STDOUT
   config.formatter = :documentation # Use the specified formatter
-  config.log_level = :error         # Avoid deprecation notice SPAM
+  config.log_level = :warn          # Avoid deprecation notice SPAM
 
   config.platform = 'ubuntu'        # Avoid warnings in ChefSpec
   config.version = '18.04'          # Avoid warnings in ChefSpec
```
Comment (damacus) [inline spec/spec_helper.rb:20]:
```suggestion
  config.version = '20.04'          # Avoid warnings in ChefSpec
  
  
```

**Review by damacus — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.0.0](https://github.com/sous-chefs/chef-splunk/releases/tag/8.0.0)

**Pull request merged**